addons/common/fnc_removePerFrameHandler.sqf

```diff
@@ -29,13 +29,19 @@ if (_handle < 0 || {_handle >= count GVAR(PFHhandles)}) exitWith {};
 [{
     params ["_handle"];
 
-    GVAR(perFrameHandlerArray) deleteAt (GVAR(PFHhandles) select _handle);
-    GVAR(PFHhandles) set [_handle, nil];
+    GVAR(perFrameHandlerArray) set [GVAR(PFHhandles) select _handle select 0, {}];
 
-    {
-        _x params ["", "", "", "", "", "_handle"];
-        GVAR(PFHhandles) set [_handle, _forEachIndex];
-    } forEach GVAR(perFrameHandlerArray);
+    [{
+        params ["_handle"];
+
+        GVAR(perFrameHandlerArray) deleteAt (GVAR(PFHhandles) select _handle);
```

Comment:
@commy2 Yeah reading again after a sleep I don't think it would break anymore (since this line becomes `deleteAt nil` on any duplicated calls - which I assume won't error?).

Reply:
The only exception would be, if you were to remove the same handle twice in the same frame and also addEventHandler in the next frame, would it be possible that the same index (`_handle`) could be refilled in between the two `execNextFrame` calls? Resulting in the newly added EH being deleted immediately.

Reply:
This is all changed now.